base/sort.jl

```diff
@@ -866,7 +866,8 @@ end
     sortperm!(ix, v; alg::Algorithm=DEFAULT_UNSTABLE, lt=isless, by=identity, rev::Bool=false, order::Ordering=Forward, initialized::Bool=false)
 
 Like [`sortperm`](@ref), but accepts a preallocated index vector `ix`.  If `initialized` is `false`
-(the default), `ix` is initialized to contain the values `1:length(v)`.
+(the default), `ix` is initialized to contain the values `1:length(v)`. Note that if `initialized` is `true`,
+and `ix` is not properly initialized, julia is likely to crash.
```

Comment:
>(the default), `ix` is initialized to contain the values `collect(eachindex(v))`, typically `1:length(v)`. If `initialized` is true, then `ix` is assumed to be an arbitrary permutation of `collect(eachindex(v))`. If this assumption is violated by `ix` not being a permutation, then `sortperm` will return wrong results. If `ix` contains out-of-bounds indices then julia may segfault, if you are lucky; or encode available secrets like passphrases in the result, if you are unlucky.

Ok, just joking. But the info I was missing was that `ix` wants to be initialized to an arbitrary permutation.

>If this assumption is violated by `ix` not being a permutation, then `sortperm` will return wrong results; if it contains oob indices then sortperm will read oob memory.